internal(replay): Add IgnoreWrapper class for hybrid SDKs

krystofwoldrich opened the pull request:
## :scroll: Description

This PR adds `IgnoreWrapper` functionality to Replay Redaction logic.

Why is this needed? In RN JS we don't have access to the native view instances directly. So we create a native wrapper class which we can use from JS.

- https://github.com/getsentry/sentry-react-native/pull/4224

## :green_heart: How did you test it?
rn sample app, unit and integration tests

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

#skip-changelog

brustolin reviewed (commented):
Just a few code improvement suggestions: In Swift, it’s better to use `guard` for simple early return, since the compiler will enforce the early return.

Apart from that, it’s all good.

krystofwoldrich commented:
@brustolin Ready for a second pass. I fixed the guards and added RedactContainer. 

To fix the issue where RedactContainer would be overwritten by IgnoreContainer in the following setup IgnoreContainer -> RedactContainer -> 'This should be redacted'

brustolin commented on Sources/Swift/Tools/UIRedactBuilder.swift:
```diff
@@ -178,9 +202,28 @@ class UIRedactBuilder {
-    
+
+    private func shouldIgnoreParentContainer(_ view: UIView) -> Bool {
+        if isRedactContainerClass(type(of: view)) {
+            return false
+        }
+
+        guard let parent = view.superview  else { return false }
+        return isIgnoreContainerClass(type(of: parent))
```
m: Try to use `guard` as much as possible for early return.

```suggestion
        guard !isRedactContainerClass(type(of: view)),  let parent = view.superview  else { return false }
        return isIgnoreContainerClass(type(of: parent))
```

brustolin reviewed (approved):
Thanks for renaming the API
